### PR TITLE
Python request logger example component deprication and removal

### LINKED ISSUE #4013: Removal & depreciation of request logger component
We are looking to depreciate and remove the request logger component as the python request logger was initially created as an example that can be used to show the functionality in the service orchestrator to push inputs and outsputs to external URLs, and show how the request can be inserted into an elasticsearch store.

As further protocols have been added and further data type schemas have been added the complexity of this component has grown significantly but it has also not improved in its reliability and best practices. We are looking to depricate and remove this component, and instead provide an example for users to showcase how they can use the request-forwarding functionality of the service orchestrator to integrate with other systems respectively.

### PR DESCRIPTION
**Pull request by axsaucedo**
Fixes #4013

- [x] Removal of Python request logger example component
- [x] Removal of documentation resources referncing python logger
- [x] Removal of e2e tests including request logger component
- [x] Outlining in upgrading documentation

The extra example using message dumper or fluentd will be tackled via #4033

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **axsaucedo**
You can assign the PR to them by writing `/assign @axsaucedo` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by SachinVarghese** [comment]
LGTM

**Comment by axsaucedo** [comment]
Thanks @SachinVarghese - any required changes before merge @michaelcheah?

**Review by michaelcheah — APPROVED** [review]

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *[michaelcheah](https://github.com/SeldonIO/seldon-core/pull/4016#pullrequestreview-933237493 "Approved")*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **axsaucedo**
You can assign the PR to them by writing `/assign @axsaucedo` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: doc/source/reference/upgrading.md
Comment (axsaucedo) [reply doc/source/reference/upgrading.md:13]: Nice one

**Comment by axsaucedo** [comment]
Nice one - thanks @SachinVarghese @michaelcheah, merging
/approve

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[axsaucedo](https://github.com/SeldonIO/seldon-core/pull/4016#issuecomment-1091291745 "Approved")*, *[michaelcheah](https://github.com/SeldonIO/seldon-core/pull/4016#pullrequestreview-933237493 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [axsaucedo]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[axsaucedo](https://github.com/SeldonIO/seldon-core/pull/4016#issuecomment-1091291745 "Approved")*, *[michaelcheah](https://github.com/SeldonIO/seldon-core/pull/4016#pullrequestreview-933237493 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [axsaucedo]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
